src/views/Users/ViewUser/index.jsx

```diff
@@ -115,79 +122,98 @@ function ViewUser({ isNewUser, ...props }) {
-
-            p.name = sc.permission;
-            p.sc = clone(sc);
-            p.sc.name = sc.permission;
-            delete p.sc.permission;
-            permissions.push(p);
+            }));
+      const permissions =
+        userdata === null
+          ? []
+          : Object.keys(userdata.data.data.permissions).map(name => {
+              const details = userdata.data.data.permissions[name];
+              const sc = scheduledChanges.data.data.scheduled_changes.filter(
+                sc => sc.username === existingUsername && sc.permission === name
+              );
+              const permission = {
+                name,
+                options: details.options || { products: [], actions: [] },
+                data_version: details.data_version,
+                sc: null,
+                metadata: {
+                  isAdditional: false,
+                  productText: '',
+                  actionText: '',
+                },
+              };
+
+              // Copy any scheduled changes associated with an existing
+              // permission into the list of permissions.
+              if (sc.length > 0) {
+                // The backend refers to permission names as the 'permission'
+                // attribute of an object, but we prefer to call it 'name', so
+                // we have to adjust that.
+                sc[0].name = sc[0].permission;
+                delete sc[0].permission;
+
+                // Array destructuring is not very readable here.
+                // eslint-disable-next-line prefer-destructuring
+                permission.sc = sc[0];
+
+                if (!permission.sc.options) {
+                  permission.sc.options = { products: [], actions: [] };
+                }
+              }
+
+              return permission;
+            });
+
+      // Scheduled inserts don't have an existing permission to associate
+      // with, so we need to create an empty one, and add to it.
+      scheduledChanges.data.data.scheduled_changes.forEach(sc => {
+        if (sc.change_type === 'insert' && sc.username === existingUsername) {
+          const p = getEmptyPermission();
+
+          p.name = sc.permission;
+          p.sc = clone(sc);
+          p.sc.name = sc.permission;
+          delete p.sc.permission;
+
+          if (!p.sc.options) {
+            p.sc.options = { products: [], actions: [] };
           }
-        });
-
-        setUsername(userdata.data.data.username);
-        setRoles(roles);
-        setOriginalRoles(clone(roles));
-        setPermissions(permissions);
-        setOriginalPermissions(clone(permissions));
-        setProducts(productdata.data.data.product);
-        setRequiredSignoffs(rs.data.data.required_signoffs);
+
+          permissions.push(p);
+        }
       });
-    }
+
+      if (!isNewUser && permissions.length === 0) {
+        setUserError('User does not exist!');
+
+        return;
+      }
+
+      setUsername(existingUsername);
```

Comment:
It would be nice to have `setUsername` come before L203 so that when a user tries to view a username that doesn't exist, the UI will have the username field prefilled.

_Example: Navigating to http://localhost:9000/users/abcd then clicking the "X" button of the error dialog_

<img width="1674" alt="Screen Shot 2019-08-07 at 8 25 11 PM" src="https://user-images.githubusercontent.com/3766511/62666458-84063680-b951-11e9-8a32-8a2ab5669c46.png">

Reply:
Ah, that's a good point. Will do.

Reply:
Is there any reason we can't set it at line 89 as the default value in `useState`? It's available there, so maybe that's the best spot, since it would ensure it's set regardless of any errors (including networking errors fetching data).

Reply:
I like that.